wdl-format/src/format.rs

```diff
@@ -0,0 +1,615 @@
+//! A module for formatting WDL code.
+
+use std::fmt::Write;
+
+use anyhow::Result;
+use format_state::SPACE;
+use wdl_ast::v1::Decl;
+use wdl_ast::v1::DocumentItem;
+use wdl_ast::v1::Expr;
+use wdl_ast::v1::HintsItem;
+use wdl_ast::v1::HintsSection;
+use wdl_ast::v1::InputSection;
+use wdl_ast::v1::LiteralBoolean;
+use wdl_ast::v1::LiteralFloat;
+use wdl_ast::v1::LiteralInteger;
+use wdl_ast::v1::LiteralString;
+use wdl_ast::v1::OutputSection;
+use wdl_ast::v1::StringPart;
+use wdl_ast::v1::StructDefinition;
+use wdl_ast::v1::Type;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Comment;
+use wdl_ast::Diagnostic;
+use wdl_ast::Direction;
+use wdl_ast::Document;
+use wdl_ast::Ident;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::Validator;
+use wdl_ast::Version;
+use wdl_ast::VersionStatement;
+
+mod comments;
+mod format_state;
+mod import;
+mod metadata;
+mod task;
+mod workflow;
+
+use comments::format_inline_comment;
+use comments::format_preceding_comments;
+use format_state::FormatState;
+
+/// Newline constant used for formatting.
+pub const NEWLINE: &str = "\n";
+/// String terminator constant used for formatting.
+const STRING_TERMINATOR: char = '"';
+
+/// A trait for elements that can be formatted.
+pub trait Formattable {
+    /// Format the element and write it to the buffer.
+    fn format(&self, buffer: &mut String, state: &mut FormatState) -> Result<()>;
```

Comment:
Since you _know_ the `Result` here can only really be a `std::fmt::Result`, I'd restrict the return type to that. This will remove the need to coerce all of the `write!(_)` results to `anyhow::Result` and will make the code a lot cleaner.